external/build/youtube-playlist.js

```diff
@@ -0,0 +1,238 @@
+            channel: playlist?.snippet?.channelId,
+            videos: videoRes,
+          });
+        } catch (error) {
+          console.error(error);
+          throw new Error('Error fetching the playlist data');
+        }
+      }
+    })
+  );
+
+  return data;
+}
+
+/**
+ * Fetches all the video meta data the component will need to render correctly the playlist.
+ * @param {string} id The YouTube playlist id
+ * @return {promise} A promise that resolves in the YouTube videos data
+ * needed to display the playlist component
+ */
+async function getPlaylistItemData(id) {
+  const data = [];
+  const response = await youtube.playlistItems.list({
+    part: PART,
+    playlistId: id,
+    auth: API_KEY,
+    maxResults: MAX_RESULTS,
+  });
+
+  const videos = response.data.items;
+
+  if (!videos || videos.length === 0) {
+    throw new Error('No playlist videos found');
+  }
+  for (const video of videos) {
+    data.push({
+      id: video.id,
+      title: video?.snippet?.title,
+      description: video?.snippet?.description,
+      thumbnail: video?.snippet?.thumbnails?.medium?.url,
+    });
+  }
+
+  return data;
+}
+
+/**
+ * Fetches all the channel meta data the component will need to render correctly the playlist.
+ * @param {string[]} id The YouTube channel id
+ * @return {promise} A promise that resolves in the YouTube Channel data needed
+ * to display the playlist component
+ */
+async function getChannelData(id) {
+  const data = {};
+  const response = await youtube.channels.list({
+    part: PART,
+    id: id,
+    auth: API_KEY,
+  });
+
+  const channels = response.data.items;
+
+  if (!channels || channels.length === 0) {
+    return data;
+  }
+  data.id = channels[0]?.id;
+  data.name = channels[0]?.snippet?.title;
+  data.description = channels[0]?.snippet?.description;
+  data.thumbnail = channels[0]?.snippet?.thumbnails?.medium?.url;
+
+  return data;
+}
+
+/**
+ * Checks if the data is stale and needs to be refetched
+ *
+ * @return {promise} A promise that resolves in a boolean to indicate if the data is stale or not
+ */
+async function checkDataTimestamp() {
+  const storage = new Storage();
+  try {
+    const file = storage
+      .bucket('external-dcc-data')
+      .file('youtube-playlist.json');
+    const stream = file.createReadStream();
+    let jsonFile = '';
+
+    stream.on('data', chunk => {
+      jsonFile += chunk.toString();
+    });
+
+    stream.on('end', () => {
+      const jsonObject = JSON.parse(jsonFile);
+      if (jsonObject['timestamp']) {
+        const dataTimestamp = jsonObject['timestamp'];
+        if (currentTimestamp - dataTimestamp < FETCH_INTERVAL) {
+          return false;
+        }
+      }
+      return true;
+    });
+
+    stream.on('error', error => {
+      console.error('Error reading the file', error);
+      return true;
+    });
+
+    return true;
+  } catch (error) {
+    console.error('Error reading file, refetching from YouTube', error);
+    return true;
+  }
```

Comment:
This can be heavily simplified 🙂  you can use `getMetadata` instead. See https://github.com/googleapis/nodejs-storage/blob/0501127f14deb7403676c93fcb53eb302fe618b7/samples/getMetadata.js#L45-L67

This saves us from handling the stream, parsing the JSON, ...